### ISSUE TITLE
SSE: first message contains only colons

### ISSUE THREAD

**Issue by jpineaufr** [nestjs/nest#8877]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When i use SSE feature, the first message received by client contains only colons(":"). It is not blocking. But this does not seem to respect the SSE standard in the absence of the "data" field.

My problem results of the file packages/core/router/sse-stream.ts, pipe (line.89) :
```
destination.write(':\n');
```

I don't understand the role of this line. Could someone help me understand its role ? Is there any solution to avoid this first message ?

This problem is also present on the SSE sample. 
```
@Sse('sse')
  sse(): Observable<MessageEvent> {
    return interval(1000).pipe(
      map((_) => ({ data: { hello: 'world' } } as MessageEvent)),
    );
  }
```

Thanks

### Minimum reproduction code

https://github.com/nestjs/nest/tree/master/sample/28-sse

### Steps to reproduce

Launch the SSE sample.
Record of raw data of the SSE client
The problem is ":\n" after HTTP headers.
```
HTTP/1.1 200 OK
X-DNS-Prefetch-Control: off
Expect-CT: max-age=0
X-Frame-Options: SAMEORIGIN
Strict-Transport-Security: max-age=15552000; includeSubDomains
X-Download-Options: noopen
X-Content-Type-Options: nosniff
X-Permitted-Cross-Domain-Policies: none
Referrer-Policy: no-referrer
X-XSS-Protection: 0
Content-Type: text/event-stream
Connection: keep-alive
Cache-Control: private, no-cache, no-store, must-revalidate, max-age=0, no-transform
Pragma: no-cache
Expire: 0
X-Accel-Buffering: no
Date: Thu, 06 Jan 2022 17:32:16 GMT
Transfer-Encoding: chunked

2
:

1f
id: 1
data: {"hello":"world"}


1f
id: 2
data: {"hello":"world"}
```

### Expected behavior

Is the line 89 is required in the file packages/core/router/sse-stream.ts ?
function pipe 
```
destination.write(':\n');
```
Maybe just use 
```
destination.write('\n');
```

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

8.2.3

### Packages versions

```json
"@nestjs/core": "^8.0.0",
```

### Node.js version

16.3.0

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

**Comment by Tony133** [comment]
Hi @jpineaufr, to get the example to work correctly, do the browser url this:  `http://127.0.0.1:3000/` and you should see this after a few seconds:
<img width="1218" alt="screenshot" src="https://user-images.githubusercontent.com/11542387/148465107-3f5d4ad4-56c2-4978-9dda-a1307822a30e.png">

**Comment by jpineaufr** [comment]
Hi @Tony133 , thanks for your answer.
Many client implementations ignore invalid events. So if you test this problem in browser with "EventSource" implementation, you cannot observe the problem.

I faced the problem with a implementation of a SSE client based on a socket. This library implements a strict interpretation of SSE event and fails on incomplete event like the first event sent by the SSE implementation of NestJS. I fixed the library to ignore invalid event and now it works with a NestJS API.

This first incorrect event is specific to the NestJS implementation. This ticket is just a suggestion for improving the SSE implementation. It is not blocking.
